core/shared/src/main/scala/io/circe/Decoder.scala

```diff
@@ -112,6 +112,20 @@ trait Decoder[A] extends Serializable { self =>
       a <- this(a1)
       b <- x(a2)
     } yield (a, b)
+
+  /**
+   * Create a new decoder that performs some operation on the incoming JSON before decoding.
+   */
+  final def after(f: HCursor => ACursor): Decoder[A] = Decoder.instance(c => self.tryDecode(f(c)))
+
```

Comment:
Yeah, now it's "decode _after_ processing the incoming JSON" and `emap` for the other `flatMap`-like thing.